subprojects/messaging/src/test/groovy/org/gradle/internal/serialize/DefaultSerializerRegistrySpec.groovy

```diff
@@ -0,0 +1,279 @@
+    def numThreads = 10
+    List<Class<?>> serializableClasses = [
+        Serializable1, Serializable2, Serializable3, Serializable4, Serializable5, Serializable6, Serializable7, Serializable8, Serializable9, Serializable10,
+        Serializable11, Serializable12, Serializable13, Serializable14, Serializable15, Serializable16, Serializable17, Serializable18, Serializable19, Serializable20,
+        Serializable21, Serializable22, Serializable23, Serializable24, Serializable25, Serializable26, Serializable27, Serializable28, Serializable29, Serializable30,
+        Serializable31, Serializable32, Serializable33, Serializable34, Serializable35, Serializable36, Serializable37, Serializable38, Serializable39, Serializable40,
+        Serializable41, Serializable42, Serializable43, Serializable44, Serializable45, Serializable46, Serializable47, Serializable48, Serializable49, Serializable50,
+        Serializable51, Serializable52, Serializable53, Serializable54, Serializable55, Serializable56, Serializable57, Serializable58, Serializable59, Serializable60,
+        Serializable61, Serializable62, Serializable63, Serializable64, Serializable65, Serializable66, Serializable67, Serializable68, Serializable69, Serializable70,
+        Serializable71, Serializable72, Serializable73, Serializable74, Serializable75, Serializable76, Serializable77, Serializable78, Serializable79, Serializable80,
+        Serializable81, Serializable82, Serializable83, Serializable84, Serializable85, Serializable86, Serializable87, Serializable88, Serializable89, Serializable90,
+        Serializable91, Serializable92, Serializable93, Serializable94, Serializable95, Serializable96, Serializable97, Serializable98, Serializable99, Serializable100
+    ]
+
+    def latch = new CountDownLatch(1)
+
+    @Subject
+    def defaultSerializerRegistry = new DefaultSerializerRegistry()
+
+    @Timeout(15)
+    def "invoking DefaultSerializerRegistry from multiple threads"() {
+        given:
+        def conditions = new AsyncConditions(numThreads)
+        numThreads.times { threadId ->
+            Thread.startDaemon("DefaultSerializerRegistry Test #$threadId") {
+                conditions.evaluate {
+                    serializableClasses.forEach { theClass ->
+                        try {
+                            defaultSerializerRegistry.useJavaSerialization(theClass)
+                            defaultSerializerRegistry.register(theClass, Mock(Serializer))
+                            Thread.sleep(1)
+                        } finally {
+                            latch.countDown()
+                        }
+                    }
+                }
+            }
+        }
+        when:
+        latch.await(5, TimeUnit.SECONDS)
+        10_000.times {
+            defaultSerializerRegistry.build(SerializableParent)
+        }
+        then:
+        noExceptionThrown()
+        conditions.await(5.0)
+        where:
+        i << (0..9)
```

Comment:
Is this just to run the same test 10 times to get different thread execution patterns?  I just searched for a `@Repeat` annotation or something, and it seems like this is actually the recommended way to do this.  Maybe name the variabe `testIteration` or something, for clarity - I was looking for an `i` which didn't exist.